crates/sui-core/src/authority.rs

```diff
@@ -1334,6 +1335,17 @@ impl AuthorityState {
                     .into());
                 }
 
+                if enforce_live_input_objects && live_object.version() < obj_ref.1 {
```

Comment:
should this be live_object.version() != obj_ref.1

Reply:
The live_object.version() > obj_ref.1 case has already been checked, and will be checked regardless of enforce_live_input_objects.